src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -452,6 +455,23 @@ class Peer(
         input.send(cmd)
     }
 
+    /**
+     * This function blocks until the peer is connected and existing channels have been fully reestablished.
+     */
+    private suspend fun waitForPeerReady() {
+        // In theory we would only need to verify that no channel is in state Offline/Syncing, but there is a corner
+        // case where a channel permanently stays in Syncing, because it is only present locally, and the peer will
+        // never send a channel_reestablish (this happens e.g. due to an error at funding). That is why we consider
+        // the peer ready if "all channels are synced" OR "peer has been connected for 10s".
+        connectionState.first { it is Connection.ESTABLISHED }
```

Comment:
We lack a metadata from the peer allowing us to tell well our channels are fully synced. In the meantime, this workaround should work.